colink/p2p_inbox.py

```diff
@@ -0,0 +1,226 @@
+        )
+        if not inbox.addr:
+            inbox = None
+        cl.vt_p2p_ctx.remote_inboxes[receiver.user_id] = inbox
+    remote_inbox = cl.vt_p2p_ctx.remote_inboxes.get(receiver.user_id, "")
+    if remote_inbox:
+        ctx = ssl.SSLContext(ssl.PROTOCOL_TLS_CLIENT)
+        cert = x509.load_der_x509_certificate(remote_inbox.tls_cert)
+        cert_file = NamedTemporaryFile()
+        cert_file.write(cert.public_bytes(Encoding.PEM))  # conver DER format to PEM
+        cert_file.seek(0)
+        ctx.load_verify_locations(cert_file.name)
+        cert_file.close()
+        ctx.check_hostname = False
+        stripped_addr = remote_inbox.addr.strip("https://")
+        conn = HTTPSConnection(stripped_addr, context=ctx)
+        headers = {
+            "user_id": cl.get_user_id(),
+            "key": key,
+            "token": remote_inbox.vt_jwt,
+        }
+        conn.request("POST", "/post", payload, headers)
+        try:
+            response = conn.getresponse()
+            if response.getcode() != Status_OK:
+                raise Exception(f"Remote inbox: error {response.getcode()}")
+        except Exception as e:
+            raise e
+    else:
+        raise Exception("Remote inbox: not available")
+
+
+def _get_variable_p2p(cl, key: str, sender: CL.Participant) -> bytes:
+    # send inbox information to the sender by remote_storage
+    if not sender.user_id in cl.vt_p2p_ctx.has_configured_inbox:
+        # create inbox if it does not exist
+        if cl.vt_p2p_ctx.public_addr and cl.vt_p2p_ctx.has_created_inbox == False:
+            cl.vt_p2p_ctx.inbox_server = VTInboxServer()
+            cl.vt_p2p_ctx.has_created_inbox = True
+        # generate vt_inbox information for the sender
+        if cl.vt_p2p_ctx.public_addr:
+            jwt_secret = cl.vt_p2p_ctx.inbox_server.jwt_secret
+            vt_jwt = jwt.encode(
+                {"user_id": sender.user_id}, jwt_secret, algorithm="HS256"
+            )
+            vt_inbox = VTInbox(
+                f"https://{cl.vt_p2p_ctx.public_addr}:{cl.vt_p2p_ctx.inbox_server.port}",
+                vt_jwt,
+                cl.vt_p2p_ctx.inbox_server.tls_cert,
+            )
+        else:
+            vt_inbox = VTInbox("", "", b"")
+        vt_inbox_vec = json.dumps(
+            {
+                "addr": vt_inbox.addr,
+                "vt_jwt": vt_inbox.vt_jwt,
+                "tls_cert": list(vt_inbox.tls_cert),
+            }
+        )
+        cl.set_variable_with_remote_storage(
+            "inbox", bytes(vt_inbox_vec, encoding="utf-8"), [sender]
+        )
+        cl.vt_p2p_ctx.has_configured_inbox.add(sender.user_id)
+    if cl.vt_p2p_ctx.public_addr == "":
+        raise Exception("Remote inbox: not available")
+    tx = Queue()
+    inbox_server = cl.vt_p2p_ctx.inbox_server
+    data = inbox_server.data_map.get((sender.user_id, key), "")
+    if data:
+        return data
+    inbox_server.notification_channels[(sender.user_id, key)] = tx
```

Comment:
Also, I think maybe it's better if you set up the channel anyway - removing three lines above